winlogbeat/eventlog/eventlog.go

```diff
@@ -82,39 +84,63 @@ type Record struct {
 func (e Record) ToEvent() beat.Event {
 	m := common.MapStr{
 		"type":          e.API,
-		"log_name":      e.Channel,
-		"source_name":   e.Provider.Name,
-		"computer_name": e.Computer,
 		"record_number": strconv.FormatUint(e.RecordID, 10),
-		"event_id":      e.EventIdentifier.ID,
 	}
 
+	m.Put("event.kind", "event")
+	m.Put("event.dataset", fmt.Sprintf("%v.%v", e.API, strings.ToLower(e.Channel)))
+	m.Put("log_name", e.Channel)
+	m.Put("source_name", e.Provider.Name)
+  // Why not keep the Windows naming for those?
+	m.Put("channel", e.Channel)
+	m.Put("provider_name", e.Provider.Name)
+	m.Put("event.code", e.EventIdentifier.ID)
+	addOptional(m, "event.action", e.Task)
+
+	m.Put("host.hostname", e.Computer)
+	m.Put("event.created", time.Now())
+
 	addOptional(m, "xml", e.XML)
 	addOptional(m, "provider_guid", e.Provider.GUID)
 	addOptional(m, "version", e.Version)
-	addOptional(m, "level", e.Level)
+	addOptional(m, "log.level", strings.ToLower(e.Level))
 	addOptional(m, "task", e.Task)
 	addOptional(m, "opcode", e.Opcode)
 	addOptional(m, "keywords", e.Keywords)
 	addOptional(m, "message", sys.RemoveWindowsLineEndings(e.Message))
-	addOptional(m, "message_error", e.RenderErr)
+	addOptional(m, "error.message", e.RenderErr)
 
 	// Correlation
 	addOptional(m, "activity_id", e.Correlation.ActivityID)
 	addOptional(m, "related_activity_id", e.Correlation.RelatedActivityID)
 
 	// Execution
-	addOptional(m, "process_id", e.Execution.ProcessID)
-	addOptional(m, "thread_id", e.Execution.ThreadID)
+	addOptional(m, "process.pid", e.Execution.ProcessID)
+	addOptional(m, "process.thread.id", e.Execution.ThreadID)
 	addOptional(m, "processor_id", e.Execution.ProcessorID)
 	addOptional(m, "session_id", e.Execution.SessionID)
 	addOptional(m, "kernel_time", e.Execution.KernelTime)
 	addOptional(m, "user_time", e.Execution.UserTime)
 	addOptional(m, "processor_time", e.Execution.ProcessorTime)
 
+	// debugf("EventData field inspection %v", e.EventData.TargetUserName)
+	// debugf("EventData inspection full %v", e.EventData)
+	// addOptional(m, "user_dbg.name", e.EventData.TargetUserName)
+
+  // Additional mapping ideas
+  // user.id
+  // - event_data.TargetUserSid (S-1-5-21-3541430928-2051711210-1391384369-1001)
```

Comment:
Depending on anything in `event_data` is tricky because the value and meaning of these fields change between sources and even event_ids. But it's possible to do if we research the event manifests and add some conditionals.

I do think I agree that that the current values for `user.{name,id}` are not correct for ECS and correlation purposes. In most cases these values correspond to the user that the logging process is running as and not to the user that the activity is reporting on. In many cases the user.name is SYSTEM and that's not helpful.

The `process_id` is the PID of the logging process and again this isn't helpful for when you are trying to find events reported about a process. I think we need a place to put this information that's not `process.*` (or simply drop it).

[cutting it short, meeting time]